src/lib.rs

```diff
@@ -463,14 +471,185 @@ impl<'a> From<LanguageTag<String>> for LanguageTag<Cow<'a, str>> {
+                unreachable!("This is a bug. en-US is supposed to be a valid LangTag! Please report this bug to https://github.com/oxigraph/oxilangtag")
+            }
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl DerefMut for SerdeLanguageTag {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.lang_tag
+    }
+}
+
+#[cfg(feature = "serde")]
+impl From<LanguageTag<String>> for SerdeLanguageTag {
+    fn from(lang_tag: LanguageTag<String>) -> Self {
+        SerdeLanguageTag {
+            lang_tag
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl Into<String> for SerdeLanguageTag {
+    fn into(self) -> String {
+        self.lang_tag.into()
+    }
+}
+
+#[cfg(feature = "serde")]
+impl From<String> for SerdeLanguageTag {
+    fn from(s: String) -> Self {
+        match LanguageTag::parse(s) {
+            Ok(lang_tag) => {
+                lang_tag.into()
+            }
+            Err(_) => {
+                SerdeLanguageTag::default()
+            }
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl Serialize for SerdeLanguageTag {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        self.tag.serialize(serializer)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> Deserialize<'de> for SerdeLanguageTag {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error> where D: Deserializer<'de> {
+        match LanguageTag::parse(String::deserialize(deserializer).unwrap_or_default()) {
+            Ok(lt) => {
+                println!("a");
+                Ok(lt.into())
+            }
+            Err(_) => {
+                println!("b");
+                Ok(SerdeLanguageTag::default())
+            }
+        }
+    }
+}
+
+#[cfg(feature = "serde")]
+impl Display for SerdeLanguageTag {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.lang_tag)
+    }
+}
+
+#[cfg(feature = "serde")]
+impl AsRef<str> for SerdeLanguageTag {
+    fn as_ref(&self) -> &str {
+        self.lang_tag.as_ref()
+    }
+}
+
+#[cfg(feature = "serde")]
+impl Borrow<str> for SerdeLanguageTag {
+    fn borrow(&self) -> &str {
+        self.lang_tag.borrow()
+    }
+}
+
+#[cfg(feature = "serde")]
+impl FromStr for SerdeLanguageTag {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(
+            match LanguageTag::parse(s.to_string()) {
+                Ok(lt) => lt.into(),
+                Err(_) => SerdeLanguageTag::default()
+            }
+        )
+    }
+}
+
 /// An error raised during [`LanguageTag`](struct.LanguageTag.html) validation.
 #[derive(Debug)]
 pub struct LanguageTagParseError {
     kind: TagParseErrorKind,
 }
 
-impl fmt::Display for LanguageTagParseError {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+impl Display for LanguageTagParseError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
```

Comment:
I find it more lenient to have the `fmt::` namespace everywhere to avoid mixing stuff with and without the namespace. But it's bike-shedding so feel free to ignore my comment.

Reply:
I got a bunch of errors when I wad compiling, rustc wad complaining that it was an "unnecessary qualifier"

Reply:
It's because `use std::fmt::{Display, Formatter};` at the beginning of the file. If you remove this line, it should not raise any warning.